### DISCUSSION TITLE
Concatenation of comic panels- different aspect ratios

### DISCUSSION THREAD

**Discussion by Tok90** [ImageMagick/ImageMagick#7802] [Help]
### ImageMagick version

7.1.1-41 

### Operating system, version and so on

MacOS Sequoia 15.1

### Description

Hello,

I have four comic panels [sample.zip](https://github.com/user-attachments/files/17879621/sample.zip), which I want to concatenate in pairs. 

That is why I wrote a command 
`magick montage -mode concatenate -tile 1x2 *.jpg out@_%d.jpg`. 

The result after running my command is below.

**[New paired panel 1. GOOD Result]** 
![out@_1](https://github.com/user-attachments/assets/fb57dd16-dcb2-417e-a00f-0c768b9a9529)

### [New paired panel 2. BAD Result]
![out@_2](https://github.com/user-attachments/assets/d8835abe-a2cc-4ae4-8954-a252f2d22908)

**The Issue:**
I don't want to concatenate _landscape_ images (Height<Weight); marked above as a bad result.
I want to concatenate _portrait images_ only (Height>Weight).

The landscape images **should be included** in the end output but omitted from the concatenation. That also means that many portrait images will remain alone (if the following/previous picture is in landscape mode).

### The wanted result:

**[New paired panel 1. GOOD Result]** 
![out@_1](https://github.com/user-attachments/assets/fb57dd16-dcb2-417e-a00f-0c768b9a9529)

**[New panel 2(omitted landscape). GOOD Result]** 
![3](https://github.com/user-attachments/assets/f4d84486-9ff1-4a4f-822d-f55cbd884fd1)

> _Could you help me write a command that solves the issue above?_

It will be run in a folder against hundreds of images (comic panels in different aspect ratios).

Thank you in advance, and all the best.

**Comment by snibgo** [comment]
I suspect you need to use a shell script. For example, this could loop through all the images. At each landscape image, it creates a tiny 1x1 pixel image, with the filename such that it will be next in alphabetical order.

**Reply by Tok90** [reply]
Thanks for the suggestion. Are there any solutions available?
I don't think I ll manage to write it on my own.

**Reply by fmw42** [reply]
Test the aspect ratio of each image first by looping over each image. Filter out the portrait mode ones from the list and use the new list of only landscape image for your latter processing in pairs.
